server/tests/users.test.js

```diff
@@ -45,11 +46,166 @@ describe("POST /api/v1/users/register", () => {
         done();
       });
   });
+}); 
+/*
+ * @route   POST api/v1/users/register
+ * @params  {name, email}
+ * @desc    Register user Missing 1 Param should send 400
+ * @access  Public
+ */
+describe("POST /api/v1/users/register", () => {
+  const body = {
+    name: "Johnny Doe",
+    email: "JohnDoeMissingParm@Gmail.com"
+
+  };
+  it("Should Respond With 400 Created", done => {
+    request
+      .post("/api/v1/users/register")
+      .send(body)
+      .set("Accept", "application/json")
+      .expect(400)
+      .end(err => {
+        if (err) return done(err);
+        done();
+      });
+  });
 });
+/*
+ * @route   POST api/v1/users/register
+ * @params  {name, email, password}
+ * @desc    Register user using same email should send 400
+ * @access  Public
+ */
+describe("POST /api/v1/users/register", () => {
+  const body = {
+    name: "Johnny Doe",
+    email: "JohnDoe@Gmail.com",
+    password: "123456"
 
-// Test -> Register Missing 1 Param SHOULD SEND 400
-// Test -> Register with existing email SHOULD SEND 400
-// Test -> User Login: Success
-// Test -> User Login: Fail Incorrect Password SEND 400
+  };
+  it("Should Respond With 400 Created", done => {
+    request
+      .post("/api/v1/users/register")
+      .send(body)
+      .set("Accept", "application/json")
+      .expect(400)
+      .end(err => {
+        if (err) return done(err);
+        done();
+      });
+  });
+});
+/*
+ * @route   POST api/v1/users/login
+ * @params  {email, password}
+ * @desc    Tests successful login, should send 200
+ * @access  Public
+ */
+describe("POST /api/v1/users/login", () => {
+  const body = {
+    email: "JohnDoe@Gmail.com",
+    password: "123456"
+  };
+  it("Should Respond With Success", done => {
+    request
+      .post("/api/v1/users/login")
+      .send(body)
+      .set("Accept", "application/json")
+      .expect(200)
+      .end((err,res) => {
+        if (err) return done(err);
+        else
+        {
+          token =res.body.token;
+        }
+        done();
+      });
+  });
+});
+/*
+ * @route   POST api/v1/users/login
+ * @params  {email, password}
+ * @desc    Test failed login should send 400
+ * @access  Public
+ */
+describe("POST /api/v1/users/login", () => {
+  const body = {
+    email: "JohnDoe@Gmail.com",
+    password: "123454"
+  };
+  it("Should Respond With Failure", done => {
+    request
+      .post("/api/v1/users/login")
+      .send(body)
+      .set("Accept", "application/json")
+      .expect(400)
+      .end(err=> {if (err)
+        return done(err);
+        done();
+      });
+  });
+});
+/*
+ * @route   GET api/v1/users/current
+ * @params  {token}
```

Comment:
If you specify "token" as a param, developers would include the token in the body of a request using "x-www-form-urlencoded like so...

![image](https://user-images.githubusercontent.com/18294827/53472367-6abd9d80-3a35-11e9-9589-7f4b895d0eda.png)

This is how developers would test routes in [Postman](https://www.getpostman.com/). Our JWT token has a special place for our Bearer Authentication in the "Authorization" tab like so ...

![image](https://user-images.githubusercontent.com/18294827/53473274-f7695b00-3a37-11e9-9834-0c719fda07ff.png)

The way HTTP/1 and HTTP/2 works, the headers that are sent to the server come with specific data within a specific format. @params specifies form data typically, while Authorization is handled differently.

Change
- @params  {token}
to
- @Authorization  {token}

Thank You!